modules/auxiliary/gather/jasmin_ransomware_dir_traversal.rb

```diff
@@ -0,0 +1,76 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Report
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Jasmin Ransomware Web Server Unauthenticated Directory Traversal',
+        'Description' => %q{
+          The Jasmin Ransomware web server contains an unauthenticated directory traversal vulnerability
+          within the download functionality. As of April 15, 2024 this was still unpatched, so all
+          versions are vulnerable. The last patch was in 2021, so it will likely not ever be patched.
+        },
+        'References' => [
+          ['CVE', '2024-30851'],
+          ['URL', 'https://github.com/chebuya/CVE-2024-30851-jasmin-ransomware-path-traversal-poc'],
+          ['URL', 'https://github.com/codesiddhant/Jasmin-Ransomware']
+        ],
+        'Author' => [
+          'chebuya', # discovery, PoC
+          'h00die', # metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'DisclosureDate' => '2023-04-08',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The relative URI of the Jasmin Ransomware webserver', '/']),
+        OptInt.new('DEPTH', [true, 'Depth of directory traversal to root ', 9]),
+        OptString.new('FILE', [true, 'File to retrieve', 'etc/passwd'])
```

Comment:
Could this datastore option enable the module to grab multiple files at once?

Reply:
as an `OptString` in theory if there was a `,` or other delimiter, however I don't think any dir travers modules within MSF do multiple files unless its a 'static' device (like a network appliance) where things are in a set location and certain files are known to be of strategic value